check.py

```diff
@@ -253,54 +250,31 @@ def run_one_spec_test(wast: Path, stdout=None, stderr=None):
     check_expected(actual, os.path.join(shared.get_test_dir('spec'), 'expected-output', test_name + '.log'), stdout=stdout)
 
 
-def run_spec_test_with_wrapped_stdout(output_queue, wast: Path):
+def run_spec_test_with_wrapped_stdout(wast: Path):
     out = io.StringIO()
     try:
-        ret = run_one_spec_test(wast, stdout=out, stderr=out)
+        run_one_spec_test(wast, stdout=out, stderr=out)
     except Exception as e:
+        # Serialize exceptions into the output string buffer
+        # so they can be reported on the main thread.
         print(e, file=out)
```

Comment:
I see that this PR slightly changed the behavior when there are failures. Previously exceptions would bubble up to the main thread and cause everything to stop, and now we can keep going which I think is good.

To make it a little easier to find what failed and debug, can we add an indication here whether it's a failed test, and print all of the failed tests at the end? That way the failed tests don't get lost in the terminal after lots of scrolling. I'm thinking something like return a bool here, and in the main thread, set aside failed stdouts to be printed at the end.

Reply:
I'm not sure about the relationship between tests failing and python `Exception` being thrown.   Are they the same time?

Maybe I could revert this part of the change?  Let me see.

Reply:
OK, I reverted this line and things seems to work fine.

Test failures are indeed represented as Exceptions.. we should probably find a way to capture them and continue to run other tests, but that is outside the scope of this PR

Reply:
Yup, that looks great.  We can do that as a followup maybe?

We could also add a `--fail-fast` option if the test runner doesn't already have one.